Objective-C/Tests/QueryTest.m

```diff
@@ -567,36 +567,36 @@ - (void) testJoin {
     [joinmeCopy setValue: @42 forKey: @"theone"];
     [self saveDocument: joinmeCopy];
     
-    CBLQuerySelectResult* MAIN_DOC_ID =
-        [CBLQuerySelectResult expression: [CBLQueryMeta idFrom: @"main"]];
+    CBLQueryExpression* propNum1 = [CBLQueryExpression property: @"number1" from: @"main"];
+    CBLQueryExpression* propNum2 = [CBLQueryExpression property: @"number2" from: @"main"];
+    CBLQueryExpression* propTheOne = [CBLQueryExpression property: @"theone" from: @"secondary"];
     
-    CBLQueryExpression* on = [[CBLQueryExpression property: @"number1" from: @"main"]
-                              equalTo: [CBLQueryExpression property:@"theone" from:@"secondary"]];
     CBLQueryJoin* join = [CBLQueryJoin join: [CBLQueryDataSource database: self.db as: @"secondary"]
-                                         on: on];
-    CBLQuery* q = [CBLQueryBuilder select: @[MAIN_DOC_ID]
+                                         on: [propNum1 equalTo: propTheOne]];
+    CBLQueryJoin* innerJoin;
+    innerJoin = [CBLQueryJoin join: [CBLQueryDataSource database: self.db as: @"secondary"]
```

Comment:
What is the different b/w innerJoin and join above?

Reply:
Oops! i thought about making this a innerJoin, just for using two different kind of statements(but same end results). But ended up being writing the same.

Reply:
what do you think, using inner-join and join with select and select-distinct, is that okay? or should we make it join in both cases? 

I used because of no particular reason. inner-join and join will give us same result, for the sake of using different statement i have used in this context.

Reply:
Yes, you can use a different one. Please add maybe one line comment on that as well.

Reply:
added the comment.